api/api.go

```diff
@@ -124,28 +130,40 @@ func NewBlockChainAPI(
 // wasn't synced up to a block where EIP-155 is enabled, but this behavior caused issues
 // in CL clients.
 func (b *BlockChainAPI) ChainId(ctx context.Context) (*hexutil.Big, error) {
+	start := time.Now()
```

Comment:
these should all be moved to an upper layer (server.go) possibly as a middleware.

Reply:
Do you want me to create a separate middleware for metrics specifically, or can I use existing functions?

Reply:
would be better to have separate middleware for metrics

Reply:
I can have another handler with `ServeHTTP` method implemented that wraps current server. I can also put code into existing `ServeHTTP`

Reply:
K. got it